test/signs_ui/signs/state_test.exs

```diff
@@ -1,9 +1,13 @@
 defmodule SignsUI.Signs.StateTest do
   use ExUnit.Case
+  use Phoenix.ChannelTest
   import SignsUI.Signs.State
   alias SignsUI.Signs
   alias SignsUI.Signs.Sign
 
+  @endpoint SignsUiWeb.Endpoint
+  @endpoint.subscribe("signs:all")
```

Comment:
I think these should be part of the test, or part of a `setup` block if they need to happen before multiple. Changing state via module attributes is usually not so good, since they only get evaluated once, when the module is compiled. I guess that's not such a problem for this `.exs` file, but it still feels a little weird to me.

Reply:
Well, I can't move it inside of the `test` because I get a `cannot set attribute @endpoint inside function/macro` error. I can put it inside the `describe`, though.

Reply:
I can at least move the `@endpoint.subscribe` though if you think that's sufficient.

Reply:
You can't set a module attribute in a function, so I was thinking you'd just have a line `SignsUiWeb.Endpoint.subscribe("signs:all")` at the start of the test. But defining `@endpoint` outside and then doing `@endpoint.subscribe` in the test also works, I suppose.

Reply:
Unfortunately if I get rid of `@endpoint` this causes issues for macros from `Phoenix.ChannelTest`, so it'll have to be the latter approach.